### PR TITLE
Cleanup cil

### PR DESCRIPTION
**Pull request by ParnianAlimi**
Address CRYP-192.

### CODE REVIEW

File: confidential-identity/Cargo.toml [hunk 1]
```diff
@@ -29,9 +29,8 @@ rand_core = { version = "0.5", default-features = false}
 rand = { version = "0.7", default-features = false }
 getrandom = { version = "0.1.14", default-features = false, optional = true}
 
-curve25519-dalek = { git = "https://github.com/PolymathNetwork/curve25519-dalek.git", branch = "v2-packed-simd", default-features = false, features = ["nightly"] }
+curve25519-dalek = { git = "https://github.com/PolymathNetwork/curve25519-dalek.git", branch = "v2-packed-simd", default-features = false, features = ["nightly", "alloc"] }
```
Comment (fmiguelgarcia) [inline confidential-identity/Cargo.toml:32]: This is enable iff features `alloc` is enable for this crate. See line 68.

**Review by fmiguelgarcia — CHANGES_REQUESTED** [review]

File: confidential-identity/Cargo.toml (hunk 1 above)
Comment (ParnianAlimi) [reply confidential-identity/Cargo.toml:32]: so the right way of doing it is to add it to lines 71 (no_std) and 72 (std). correct?

Comment (ParnianAlimi) [reply confidential-identity/Cargo.toml:32]: or just `std`? since wasm doesn't support `alloc` (I think)?

Comment (fmiguelgarcia) [reply confidential-identity/Cargo.toml:32]: I mean, if this crate has an `alloc` feature, any dependency with `alloc` should be enabled if and only if the general crate feature is enabled.
If you add that change, `curve25519-dalek` will have `alloc` even if this crate has `alloc` disabled, which is a bit inconsistent.

Comment (ParnianAlimi) [reply confidential-identity/Cargo.toml:32]: right, but I am using functionalities in the Dalek library (scalar multiplication) that are only available as an alloc feature.

Comment (ParnianAlimi) [reply confidential-identity/Cargo.toml:32]: This crate used to wrongly depend on bulletproofs which by default depended on Dalek's alloc feature, and that's how it was working before. Now that I have removed the dependency with bulletproofs, I need to enable alloc.

Comment (fmiguelgarcia) [reply confidential-identity/Cargo.toml:32]: In that case, `alloc` is a **mandatory** `feature` for `confidential-identity`, so you could not disable it. That means, that `no_std` and `std` should have `alloc` enable, which means:
1. `bulletproof`'s features are `["zeroize", "alloc"]`,
2. `rand_core`'s features are `["alloc"]`,
3. `rand`'s features are `["alloc"]`,
4. This crate should not have an `alloc` features, so you cannot disable with `--no-default-features`.

**Review by fmiguelgarcia — APPROVED** [review]

**Pull request merged**